SwiftReflector/MarshalEngine.cs

```diff
@@ -1229,6 +1240,89 @@ CSBaseExpression MarshalProtocolConstrained (BaseDeclaration typeContext, Functi
 
 		}
 
+		CSBaseExpression MarshalAssociatedTypeProtocolConstrained (BaseDeclaration typeContext, FunctionDeclaration wrapperFunc, CSParameter p, NamedTypeSpec swiftType)
+		{
+			// var proxy = p is ProxyType ? (InterfaceType)p : SwiftProtocolTypeAttribute.MakeProxy (typeof (proxyType), (InterfaceType)p);
+			// var proxyRefCount = StructMarshal.RetainCount ((ISwiftObject)proxy);
+			// var proxyPtr = stackalloc byte [IntPtr.Size];
+			// var proxyIntPtr = new IntPtr (proxyPtr);
+			// StructMarshal.Marshaler.ToSwift (proxy, proxyIntPtr);
+			// ...
+			// if (p is ProxyType && proxyRefCount >= StructMarshal.RetainCount ((ISwiftObject)proxy)) {
+			//    ((ISwiftObject)proxy).Dispose ();
+			// }
+
+			var protocolConstraint = ProtocolConstraintFromFunctionDeclaration (swiftType, wrapperFunc);
+			var csInterfaceTypeNTB = typeMapper.MapType (wrapperFunc, new NamedTypeSpec (protocolConstraint.ToFullyQualifiedName ()), false);
+			var csInterfaceType = csInterfaceTypeNTB.ToCSType (use) as CSSimpleType;
+			var csProxyType = BuildGenericInterfaceFromAssociatedTypes (csInterfaceType, protocolConstraint);
+
+			RequiredUnsafeCode = true;
+			var proxyID = new CSIdentifier (Uniqueify ("proxy", identifiersUsed));
```

Comment:
`Uniqueify`? Used to seeing this within the context of values in a data structure, curious about why is it needed here?

Reply:
`Uniqueify` is used to generate a symbol that is unique within a particular scope. In this case, I'm creating local variables named `proxy`, `proxyPtr` and `proxyIntPtr`. If I have more than one generic argument with a PAT constraint and no `Uniqueify` there will be a name conflict. With uniquing, we get `proxy`, `proxy0`, `proxy1`, ... Any marshaling that requires an intermediate value needs this. In addition, `Uniqueify` will also protect us from parameter names that conflict with locals as well (eg, a parameter named `proxy` etc).
`Uniqueify` is fairly st0opId but I knocked it together years ago to solve a problem and just moved on. I considered putting on my architect Underoos™ and building a Uniquing System that would do things like use a `HashSet` and maintain scopes so that I could use one uniquer across, say, all C# code, but decided that wasn't worth it.

Reply:
Whoa, didn't realize it was a custom tool that you built. Makes a lot of sense to avoid all kinds of name conflicts with a highly sophisticated Uniquing System&trade;